packages/stream/create-stream/create-stream.js

```diff
@@ -0,0 +1,114 @@
+ * @returns {string} -- stream's slug as per stream's name but without space
+ */
+function replaceWhiteSpace(text) {
+  return /\s/.test(text) ? text.split(' ').join('-') : text
+}
+
+/**
+ * @typedef Config
+ * @property {string} name -- name of stream
+ * @property {string} slug -- slug of stream
+ * @property {string} description -- description of stream
+ */
+
+/**
+ * @typedef FetchResponse
+ * @property {object} status -- A status response
+ * @property {object} data -- A return data as per endpoint
+ */
+
+/**
+ * A create stream module that should pass stream's name parameter, and user could pass the stream's slug & description parameter as well
+ *
+ * @function
+ * @param {InitializationInstance} initObject -- initialization object
+ * @param {Config} config - passing param from user including name, description, slug
+ * @returns {Promise<FetchResponse>} returns the restructured data which content status & created stream data
+ * @throws {Error}
+ */
+export const createStream = async (initObject, config) => {
+  if (!(initObject instanceof InitializationInstance) || !config) {
+    throw new TypeError(
+      'Failed to process because initialization is not valid and/or missing config. Please provide required initialization argument which is the initialization instance returned by the init() function and the configuration object'
+    )
+  } else if (typeof config !== 'object') {
+    throw new TypeError(
+      'Failed to process because config argument must be in object format'
+    )
+  } else if (config.name === null || config.name === undefined) {
+    throw new Error(
+      'Failed to create a new stream because the name of the stream is empty. Please provide a stream name'
+    )
+  } else if (typeof config.name !== 'string') {
+    throw new TypeError(
+      'Failed to create a new stream because the name of the stream is not in string format. A stream name must be in string format'
+    )
+  } else if (
+    config.slug !== null &&
+    config.slug !== undefined &&
+    typeof config.slug !== 'string'
+  ) {
+    throw new Error(
+      'Failed to create a new stream because the slug of the stream is not in string format. A slug must be in string format'
+    )
+  } else if (
+    config.description !== null &&
+    config.description !== undefined &&
+    typeof config.description !== 'string'
+  ) {
+    throw new Error(
+      'Failed to create a new stream because the description of the stream is not in string format. A description must be in string format'
+    )
+  } else {
+    let fetchResponse = await Internal.fetchHttp({
+      url: `${Internal.config.api.base_url}/${Internal.config.api.version}/streams/create`,
+      token: initObject.config.api_key,
+      method: 'POST',
+      body: {
+        name: config.name,
+        slug: config.slug || replaceWhiteSpace(config.name),
+        description: config.description || '',
+      },
+    }).catch((error) => {
+      return error
```

Comment:
Same with [this PR](https://github.com/inlivedev/inlive-js-sdk/pull/12/files#r973819622). You can handle the error here or `throw` the error and catch it on another place. But please don't return the error.

Reply:
@faiq-naufal done at commit [2b610de](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/2b610de1aaf72e6f2b31ac37f9e68b5b79402d95)

Reply:
@faiq-naufal Pardon me, I just reverted back at commit [a2fa550](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/a2fa5508621d0345ff792fb3a3828f7979fdf700) based on same as [this PR](https://github.com/inlivedev/inlive-js-sdk/pull/12#discussion_r974039900) consideration.